apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -1020,40 +1020,28 @@ function handleInlineHelpButton( calypsoPort ) {
  * @param {MessagePort} calypsoPort Port used for communication with parent frame.
  */
 function handleSiteEditorBackButton( calypsoPort ) {
+	// We use the traversal helper because the target element may be the SVG or an SVG element inside.
```

Comment:
Some interesting recursion going on here. I think I understand how/why we're doing this, but I wouldn't mind getting one extra pair of eyes on this before merging. 

Conversely, I know this is a bug fix, so maybe we release first to resolve this issue, but then I still wouldn't mind separate eyes on this after the fact to see if anyone else can anticipate any issues with this method.

Reply:
As far as performance goes, this linked list traversal doesn’t look super concerning, especially because it’s limited to, at most, two cycles ( depth = 2 ).

It's still dependent on the location of the a tag ( what if it's not a parent _or_ grandparent element, but a great grandparent instead? ), but at this point in time, it's the nature of working with open source Gutenberg